torchvision/models/detection/rpn.py

```diff
@@ -355,12 +355,9 @@ def filter_proposals(self, proposals, objectness, image_shapes, num_anchors_per_
         final_scores = []
         for boxes, scores, lvl, img_shape in zip(proposals, objectness, levels, image_shapes):
             boxes = box_ops.clip_boxes_to_image(boxes, img_shape)
-            keep = box_ops.remove_small_boxes(boxes, self.min_size)
-            boxes, scores, lvl = boxes[keep], scores[keep], lvl[keep]
             # non-maximum suppression, independently done per level
-            keep = box_ops.batched_nms(boxes, scores, lvl, self.nms_thresh)
+            keep = box_ops.batched_nms(boxes, scores, lvl, self.nms_thresh, self.post_nms_top_n)
```

Comment:
Ditto.

Reply:
Basically you need to ensure that the hyperparameters you choose will enforce that `len(boxes) >= self.post_nms_top_n`. It's not that hard: if it's not the case, just make sure to generate more boxes by increasing the resize size, decreasing the stride, etc. Also if the assert is satisfied for one image, it'll be for all of them. See Finding 5 for more detail.

Reply:
We can do it in a followup CL, but can we have a check to generate more boxes if `len(boxes) < self.post_nms_top_n`?